htdocs/adherents/class/api_memberstypes.class.php

```diff
@@ -79,14 +79,15 @@ public function get($id)
      *
      * @param string    $sortfield  Sort field
      * @param string    $sortorder  Sort order
+     * @param string    $nature     Nature of type phy, mor or both (for only both not mor or phy only)
      * @param int       $limit      Limit for list
      * @param int       $page       Page number
      * @param string    $sqlfilters Other criteria to filter answers separated by a comma. Syntax example "(t.libelle:like:'SO-%') and (t.subscription:=:'1')"
      * @return array                Array of member type objects
      *
      * @throws RestException
      */
-    public function index($sortfield = "t.rowid", $sortorder = 'ASC', $limit = 100, $page = 0, $sqlfilters = '')
+    public function index($sortfield = "t.rowid", $sortorder = 'ASC', $nature = 'all', $limit = 100, $page = 0, $sqlfilters = '')
```

Comment:
There is already a filed sqlfilters to allow to filter on the properties/values or our choice.
To avoid to add tons of filter criteria, goal is to have only one parameter to make any advanced filtering on any field of our choice.

Reply:
@eldy  for this properties, i identify 2 choices of filter: no filter & phy or mor 

with sqlfilter we have no filter or  null(=both nature) or only phy or only mor but mor or phy include null(both) value

ex type d'adhésion: personnel (value: phy + null) , professionnel (value: mor + null),  les 2 (value: null). We can't filter this with sqlfilter

Reply:
All combinations of criteria should be possible with sqlfilter. For example, you can use

(t.nature:=:'mor') or (t.nature:is:null)

Reply:
@eldy good for me! i deleted this part of code in API